### ISSUE TITLE
NFS volumes fail to mount

### ISSUE THREAD

**Issue by aadu** [kubernetes/minikube#1085]
While experimenting with [dynamic nfs volume provisioning](https://github.com/kubernetes-incubator/nfs-provisioner), I ran into the following error:

```
MountVolume.SetUp failed for volume "kubernetes.io/nfs/4b24c73a-e9f2-11e6-afb7-96bffef5a320-pvc-2c49f94e-e9f2-11e6-afb7-96bffef5a320" (spec.Name: "pvc-2c49f94e-e9f2-11e6-afb7-96bffef5a320") pod "4b24c73a-e9f2-11e6-afb7-96bffef5a320" (UID: "4b24c73a-e9f2-11e6-afb7-96bffef5a320") with: mount failed: exit status 32
Mounting command: mount
Mounting arguments: 10.0.0.58:/export/pvc-2c49f94e-e9f2-11e6-afb7-96bffef5a320 /var/lib/kubelet/pods/4b24c73a-e9f2-11e6-afb7-96bffef5a320/volumes/kubernetes.io~nfs/pvc-2c49f94e-e9f2-11e6-afb7-96bffef5a320 nfs []
Output: mount.nfs: an incorrect mount option was specified
```

**Environment**:
- **minikube version**: `v0.16.0`
- **OS**: `Darwin MacBook-Pro-2.local 16.4.0`
- **VM Driver**: `xhyve`
- **ISO version**: `minikube-v1.0.6.iso`

**How to reproduce it**:
Just follow the demo here: https://github.com/kubernetes-incubator/nfs-provisioner/tree/master/demo
```
git clone https://github.com/kubernetes-incubator/nfs-provisioner.git
cd nfs-provisioner
kubectl create -f demo/deployment.yaml
kubectl create -f demo/class.yaml
kubectl create -f demo/claim.yaml
kubectl create -f demo/nfs-busybox-rc.yaml
kubectl describe pod -l name=nfs-busybox
```

**Anything else do we need to know**:
Saw [this issue](https://github.com/kubernetes/kubernetes/issues/33447). Don't know if the discussion is helpful or not.

**Comment by mstrzele** [comment]
Maybe Kubernetes tries to use wrong NFS `mount`command?

Doing an NFS mount from the shell gives the same error:
```bash
$ minikube ssh
$ sudo mount -t nfs 192.168.64.1:/Users /Users
mount.nfs: an incorrect mount option was specified
$ mount -V                                    
mount from util-linux 2.28 (libmount 2.28.: btrfs, assert, debug)
```

When `mount` from BusyBox is used there's no such error:
```bash
$ sudo busybox mount -t nfs 192.168.64.1:/Users /Users
$ echo $?
0
```

It might be related to #1033. The NFS client was recently added there I think.

**Comment by r2d4** [comment]
Right now we're using busybox mount, we should probably switch back to old mount

**Comment by fjo-extia** [comment]
The problem comes from default version used by mount.nfs (vers=4.2) while the kernel is compiled without support for NFS 4.2.

One workaround could be to change /etc/nfsmount.conf on the iso to uncomment "Defaultvers=4" and "Nfsvers=4".

Failed default mount.
```
$ sudo mount -v 192.168.43.1:/exports /mnt
mount.nfs: timeout set for Fri Mar 17 08:54:41 2017
mount.nfs: trying text-based options 'vers=4.2,addr=192.168.43.1,clientaddr=192.168.43.54'
mount.nfs: mount(2): Invalid argument
mount.nfs: an incorrect mount option was specified
```

Possibility of mount by passing the right options.
```
$ sudo mount -v 192.168.43.1:/exports /mnt -o vers=4
mount.nfs: timeout set for Fri Mar 17 08:54:53 2017
mount.nfs: trying text-based options 'vers=4,addr=192.168.43.1,clientaddr=192.168.43.54'
192.168.43.1:/home/joulaud/nfs-export on /mnt type nfs (rw,vers=4)
```

kernel options of minikube iso:
```
$ gunzip -c /proc/config.gz | grep _NFS_V 
CONFIG_NFS_V2=y
CONFIG_NFS_V3=y
# CONFIG_NFS_V3_ACL is not set
CONFIG_NFS_V4=y
CONFIG_NFS_V4_1=y
# CONFIG_NFS_V4_2 is not set
CONFIG_NFS_V4_1_IMPLEMENTATION_ID_DOMAIN="kernel.org"
# CONFIG_NFS_V4_1_MIGRATION is not set
```

This is blocking for my use case as nfs volumes definition in kubernetes cannot specify mount options.

**Comment by fjo-extia** [comment]
Not this is related to this nfs-utils bug too: https://bugzilla.opensuse.org/show_bug.cgi?id=959211

**Comment by r2d4** [comment]
I'm for just enabling nfs v4 in the kernel. Any problems with that?

**Comment by fjo-extia** [comment]
The config option to enable is CONFIG_NFS_V4_2 which enables nfs v4.2 (nfs v4 and v4.1 are already enabled). No idea why this option was not set in the kernel so I have nothing against setting it. If it solves this problem I will be happy whatever the solution.

**Comment by aaron-prindle** [comment]
Closing as this was resolved with #1256.  The commands from the nfs-provisioner repo are working with the latest version of minikube.
